server/resource.go

```diff
@@ -127,24 +129,40 @@ func (s *Server) registerResourceRoutes(g *echo.Group) {
 			if !strings.Contains(filePath, "{filename}") {
 				filePath = path.Join(filePath, "{filename}")
 			}
-			filePath = path.Join(s.Profile.Data, replacePathTemplate(filePath, file.Filename))
-			dir, filename := filepath.Split(filePath)
+
+			hashFilename := publicID + filepath.Ext(file.Filename)
+			filePath = path.Join(s.Profile.Data, replacePathTemplate(filePath, hashFilename))
+			dir, _ := filepath.Split(filePath)
 			if err = os.MkdirAll(dir, os.ModePerm); err != nil {
 				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create directory").SetInternal(err)
 			}
-			dst, err := os.Create(filePath)
+
+			fileBytes, err := io.ReadAll(sourceFile)
 			if err != nil {
-				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create file").SetInternal(err)
+				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to read resource file").SetInternal(err)
 			}
-			defer dst.Close()
-			_, err = io.Copy(dst, sourceFile)
+
+			err = os.WriteFile(filePath, fileBytes, 0666)
```

Comment:
We don't need to modify the original filename.  It should be handled by the path template.